### PR TITLE
[SPARK-31259][CORE] Fix log message about fetch request size in ShuffleBlockFetcherIterator

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Fix incorrect log of `cureRequestSize`.

### Why are the changes needed?

In batch mode, `curRequestSize` can be the total size of several block groups. And each group should have its own request size instead of using the total size. 

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

It's only affect log.

### CODE REVIEW

**Comment by Ngone51** [comment]
ping @cloud-fan @jiangxb1987 @xuanyuanking Please take a look, thanks!

File: core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala [hunk 1]
```diff
@@ -329,27 +329,25 @@ final class ShuffleBlockFetcherIterator(
 
   private def createFetchRequest(
       blocks: Seq[FetchBlockInfo],
-      address: BlockManagerId,
-      curRequestSize: Long): FetchRequest = {
-    logDebug(s"Creating fetch request of $curRequestSize at $address "
+      address: BlockManagerId): FetchRequest = {
+    logDebug(s"Creating fetch request of ${blocks.map(_.size).sum} at $address "
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala]: isn't `curRequestSize` also the sum of input blocks size?

Comment (Ngone51) [reply core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala]: In non batch mode,  it is. But in batch mode, `blocks` here can a part of all blocks as they will be grouped by `maxBlocksInFlightPerAddress` before calling `createFetchRequest`.

**Review by dongjoon-hyun — APPROVED** [review]

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Comment by jiangxb1987** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Comment by Ngone51** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Thank you all. Merged to master/3.0.

**Comment by Ngone51** [comment]
thanks all!!

**Pull request closed without merging**